src/sentry/sentry_metrics/consumers/indexer/multiprocess.py

```diff
@@ -221,12 +186,7 @@ def join(self, timeout: Optional[float] = None) -> None:
                 timeout = 5.0
             self.__producer.flush(timeout)
 
-        if self.__callbacks:
-            logger.info(f"Committing {self.__callbacks} messages...")
-            self.__commit_function(self.__produced_message_offsets)
-            self.__callbacks = 0
-            self.__produced_message_offsets = {}
```

Comment:
We are now unconditionally clearing this map even if the commitpolicy throttled the commit call. I hope that is okay. I was initially not entirely sure what would happen if one would do this:

1. call `commit({1: 2})`, which gets throttled
2. call `(commit({2: 4})`, which gets accepted

will partition 1 be committed?

the answer seems to be yes if I read https://github.com/getsentry/arroyo/blob/eadfaa1436ac70dbeaed914737acf854ef76a5b0/arroyo/processing/processor.py#L151-L153 correctly.

Reply:
Yup it will. All changes are staged even if throttled for commit later.